api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3895,6 +3898,107 @@ public ObjectLockConfiguration getDefaultRetention(String bucketName)
   }
 
 
+  /**
+   * Applies object retention lock onto an object.
+   *
+   * </p><b>Example:</b><br>
+   * <pre>{@code minioClient.setObjectRetention("my-bucketname", "my-object", config );
+   * System.out.println("Set object retention on my-object successfully."); }</pre>
+   *
+   * @param bucketName Bucket name.
+   * @param objectName Object name.
+   * @param config     Object lock configuration.
+   *
+   * @throws InvalidBucketNameException  upon invalid bucket name is given
+   * @throws NoSuchAlgorithmException
+   *           upon requested algorithm was not found during signature calculation
+   * @throws InsufficientDataException  upon getting EOFException while reading given
+   *           InputStream even before reading given length
+   * @throws IOException                 upon connection error
+   * @throws InvalidKeyException
+   *           upon an invalid access key or secret key
+   * @throws NoResponseException         upon no response from server
+   * @throws XmlPullParserException      upon parsing response xml
+   * @throws ErrorResponseException      upon unsuccessful execution
+   * @throws InternalException           upon internal library error
+   * @throws InvalidResponseException    upon a non-xml response from server
+   */
+  public void setObjectLockRetention(String bucketName, String objectName, ObjectLockRetentionConfiguration config)
```

Comment:
We need to comply with S3 API names to avoid confusion i.e. `setObjectRetention()` and `getObjectRetention()`.  Below are right signature as per the spec.

```java
public void setObjectRetention(String bucketName, String objectName, String versionId, boolean bypassGovernanceRetention, ObjectRetentionConfiguration config)
public ObjectRetentionConfiguration getObjectRetention(String bucketName, String objectName, String versionId)
```